saleor/graphql/account/bulk_mutations/customer_bulk_update.py

```diff
@@ -0,0 +1,586 @@
+                "first_name",
+                "last_name",
+                "company_name",
+                "street_address_1",
+                "street_address_2",
+                "city",
+                "city_area",
+                "postal_code",
+                "country",
+                "country_area",
+                "phone",
+            ],
+        )
+
+        models.User.objects.bulk_update(
+            customers_to_update,
+            fields=[
+                "first_name",
+                "last_name",
+                "email",
+                "is_active",
+                "note",
+                "language_code",
+                "external_reference",
+                "updated_at",
+            ],
+        )
+
+        for customer in customers_to_update:
+            if customer in customer_instance_new_addresses_map:
+                customer.addresses.add(*customer_instance_new_addresses_map[customer])
+                # refresh customer default addresses
+                customer.default_billing_address = customer.default_billing_address
+                customer.default_shipping_address = customer.default_shipping_address
+
+            search_document = prepare_user_search_document_value(customer)
+            customer.search_document = search_document
+
+        models.User.objects.bulk_update(
+            customers_to_update,
+            fields=[
+                "default_shipping_address",
+                "default_billing_address",
+                "search_document",
+            ],
+        )
+
+        return customers_to_update, old_instances
+
+    @classmethod
+    def post_save_actions(cls, info, manager, instances, old_instances):
+        customer_events = []
+        app = get_app_promise(info.context).get()
+        staff_user = info.context.user
+
+        for updated_instance, old_instance in zip(instances, old_instances):
+            cls.call_event(manager.customer_updated, updated_instance)
+            new_email = updated_instance.email
+            new_fullname = updated_instance.get_full_name()
+
+            # Compare the data
+            has_new_name = old_instance.get_full_name() != new_fullname
+            has_new_email = old_instance.email != new_email
+            was_activated = not old_instance.is_active and updated_instance.is_active
+            was_deactivated = old_instance.is_active and not updated_instance.is_active
+
+            # Generate the events accordingly
+            if has_new_email:
+                customer_events.append(
+                    models.CustomerEvent(
+                        user=staff_user,
+                        app=app,
+                        order=None,
+                        type=CustomerEvents.EMAIL_ASSIGNED,
+                        parameters={"message": new_email},
+                    )
+                )
+                assign_user_gift_cards(updated_instance)
+                match_orders_with_new_user(updated_instance)
```

Comment:
I'm wondering if we shouldn't turn these functions into bulk ones, now each calls some DB queries and we call them in a loop.

Reply:
I agree, I think this and updating the search vector for bulk mutation should be done in Celery task for example. But we should do this in separate PR.

Reply:
That's fine, but are we tracking this work that's left in some task?

Reply:
yep, I added ticket to core backlog